apps/front/src/components/ui/formButton.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { CircularProgress } from '@mui/material'
+import ClearIcon from '@mui/icons-material/Clear'
+import { ApolloError } from '@apollo/client'
+
+type FormButtonType = {
+    loading: boolean
+    isError?: boolean | ApolloError
```

Comment:
czemu można przekazać do propsa isError ApolloError? Jak tworzysz komponenty tu rób tak, aby były maksymalnie neizależne od technologii z jakiej korzystasz. Tak samo material nie dodaje do swich propsów interface'ow z Apollo czy RTK lub jeszcze innych

Reply:
No miałem tutaj problem taki, że error który wrzucam w props ma nadane z apollo ` ApolloError | undefined ` i jak nie utypuje isError jako ` | ApolloError ` to dostaję błąd, że boolean nie jest równy otrzymanemu typowi ` ApolloError | undefined `

Reply:
Czy mam np jakoś ogarnąć w mutacji/query że onError ustawić state error boolean na true i wtedy go puszczać do FormButton? Bo obecnie przekazywałem własność z apollo ` const [createUser, { loading, error }] = useMutation(CREATE_USER, ` no i ten error ma takie typy